daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/RollbackTest.scala

```diff
@@ -194,43 +195,60 @@ class ExceptionTest extends AnyWordSpec with Matchers with TableDrivenPropertyCh
        }
       """)
 
-  val testCases = Table[String, List[Long]](
+  val testCases = Table[String, List[Tree]](
     ("expression", "expected-number-of-contracts"),
     ("create0", Nil),
-    ("create1", List(100)),
-    ("create2", List(100, 200)),
-    ("create3", List(100, 200, 300)),
-    ("create3nested", List(100, 200, 300)),
-    ("create3catchNoThrow", List(100, 200, 300)),
-    ("create3throwAndCatch", List(300)),
-    ("create3throwAndOuterCatch", List(300)),
-    ("exer1", List(100, 400, 500, 200, 300)),
-    ("exer2", List(100, 300)),
+    ("create1", List(C(100))),
+    ("create2", List(C(100), C(200))),
+    ("create3", List(C(100), C(200), C(300))),
+    ("create3nested", List(C(100), C(200), C(300))),
+    ("create3catchNoThrow", List(C(100), C(200), C(300))),
+    ("create3throwAndCatch", List[Tree](R(List(C(100), C(200))), C(300))),
+    ("create3throwAndOuterCatch", List[Tree](R(List(C(100), C(200))), C(300))),
+    ("exer1", List[Tree](C(100), X(List(C(400), C(500))), C(200), C(300))),
+    ("exer2", List[Tree](C(100), R(List(X(List(C(400))))), C(300))), // TODO: Is the X correct?
```

Comment:
What are you unsure about here? I’m probably missing something but it looks fine to me.

Reply:
@nickchapman-da is unsure. 
I am not sure about what it is unsure. 
I will review this test again.

Reply:
Looks fine for me.